ccl/multiregionccl: TestMrSystemDatabase failed

ccl/multiregionccl.TestMrSystemDatabase [failed](https://mesolite.cluster.engflow.com/invocations/default/98bc678b-abdb-4dc7-a724-b30c69b4e994?testReportRun=7&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on release-26.1 @ [61121b5fdc0ae61be17fc74c542775eb2f875e04](https://github.com/cockroachdb/cockroach/commits/61121b5fdc0ae61be17fc74c542775eb2f875e04):


```
=== RUN   TestMrSystemDatabase
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestMrSystemDatabase1849040183
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:94: cluster virtualization disabled due to issue: #156308 (expected label: C-test-failure)
    multiregion_system_table_test.go:324: -- test log scope end --
test logs left over in: outputs.zip/logTestMrSystemDatabase1849040183
--- FAIL: TestMrSystemDatabase (141.02s)
=== RUN   TestMrSystemDatabase/Sqlliveness_system_database
    multiregion_system_table_test.go:116: condition failed to evaluate within 45s: pq: unsupported comparison: bytes to crdb_internal_region
--- FAIL: TestMrSystemDatabase/Sqlliveness_system_database (45.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #159048 ccl/multiregionccl: TestMrSystemDatabase failed [unsupported comparison] [C-test-failure O-robot P-3 T-sql-foundations branch-master]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMrSystemDatabase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60516

--- a/pkg/ccl/multiregionccl/multiregion_system_table_test.go
+++ b/pkg/ccl/multiregionccl/multiregion_system_table_test.go
@@ -67,6 +67,14 @@ func TestMrSystemDatabase(t *testing.T) {
 
 	tDB := sqlutils.MakeSQLRunner(tenantSQL)
 
+	// Disable automatic stats collection to prevent a race condition during the
+	// multi-region system database conversion. The conversion changes the
+	// crdb_region column type from bytes to the crdb_internal_region enum and
+	// deletes old stats, but automatic stats collection can re-insert stats with
+	// the old column type before the async refresh completes, causing
+	// "unsupported comparison: bytes to crdb_internal_region" errors.
+	tDB.Exec(t, `SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false`)
+
 	// Generate stats for system.sqlinstances. See the "QueryByEnum" test for
 	// details.
 	tDB.Exec(t, `ANALYZE system.sqlliveness;`)
@@ -85,6 +93,7 @@ func TestMrSystemDatabase(t *testing.T) {
 
 	sDB := sqlutils.MakeSQLRunner(systemSQL)
 
+	sDB.Exec(t, `SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false`)
 	sDB.Exec(t, `ANALYZE system.sqlliveness;`)
 	sDB.Exec(t, `SET CLUSTER SETTING sql.multiregion.system_database_multiregion.enabled = true`)
 	sDB.Exec(t, `ALTER DATABASE system SET PRIMARY REGION "us-east1"`)

